internal/k8s/controller.go

```diff
@@ -1504,10 +1504,25 @@ func (lbc *LoadBalancerController) createVirtualServer(virtualServer *conf_v1alp
 	}
 
 	endpoints := make(map[string][]string)
+	externalNameSvcs := make(map[string]bool)
 
 	for _, u := range virtualServer.Spec.Upstreams {
-		endpointsKey := configs.GenerateEndpointsKey(virtualServer.Namespace, u.Service, u.Port)
-		endpoints[endpointsKey] = lbc.getEndpointsForService(virtualServer.Namespace, u.Service, int(u.Port))
+		backend := &extensions.IngressBackend{
+			ServiceName: u.Service,
+			ServicePort: intstr.FromInt(int(u.Port)),
+		}
+
+		svc, err := lbc.getServiceForIngressBackend(backend, virtualServer.Namespace)
+		if err != nil {
+			glog.V(3).Infof("Error getting service %v: %v", u.Service, err)
+		} else {
+			endpointsKey := configs.GenerateEndpointsKey(virtualServer.Namespace, u.Service, u.Port)
+			endps, external, err := lbc.getEndpointsForIngressBackend(backend, virtualServer.Namespace, svc)
```

Comment:
in case of an error here and in line 1579, could we report a warning similarly to what we do in `createIngressEx` https://github.com/nginxinc/kubernetes-ingress/blob/master/internal/k8s/controller.go#L1408